WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -232,16 +187,10 @@ public bool TryGetSpentCoinByOutPoint(OutPoint outPoint, [NotNullWhen(true)] out
 			// destroyed (spent) coins are now (unspent)
 			foreach (SmartCoin destroyedCoin in allCoins.SpentBy(txId))
 			{
-				if (SpentCoins.Remove(destroyedCoin))
-				{
-					SpentCoinsByOutPoint.Remove(destroyedCoin.Outpoint);
-					Coins.Add(destroyedCoin);
-					toAdd.Add(destroyedCoin);
-				}
+				destroyedCoin.SpenderTransaction = null;
```

Comment:
Currently on master, we are not unregistering the SpenderTransaction in case of an `Undo`. FWIU, it's a mistake. See